File: mod/1.21.x-8/neoforge/src/main/java/net/wiredtomato/waygl/neoforge/service/NeoForgePlatformService.java
package net.wiredtomato.waygl.neoforge.service;

import com.google.auto.service.AutoService;
import net.neoforged.fml.ModList;
import net.neoforged.fml.loading.FMLEnvironment;
import net.neoforged.fml.loading.FMLLoader;
import net.neoforged.fml.loading.FMLPaths;
import net.wiredtomato.waygl.core.service.PlatformService;

import java.nio.file.Path;

@AutoService(PlatformService.class)
public class NeoForgePlatformService implements PlatformService {
    @Override
    public String platform() {
        return "neoforge";
    }

    @Override
    public Boolean isModLoaded(String modId) {
        return ModList.get().isLoaded(modId);
    }

    @Override
    public Boolean isDevelopmentEnvironment() {
        return !FMLEnvironment.production;
    }

    @Override
    public String getMinecraftVersionString() {
        return FMLLoader.versionInfo().mcVersion();
    }

    @Override
    public Path getConfigDirectory() {
        return FMLPaths.CONFIGDIR.get();
    }
}

File: mod/neoforge/src/main/kotlin/net/wiredtomato/waygl/service/NeoForgePlatformService.kt
package net.wiredtomato.waygl.service

import deplatformed.ServiceImpl
import net.neoforged.fml.ModList
import net.neoforged.fml.loading.FMLEnvironment
import net.neoforged.fml.loading.FMLLoader
import net.neoforged.fml.loading.FMLPaths
import java.nio.file.Path

@ServiceImpl(PlatformService::class)
class NeoForgePlatformService : PlatformService {
    override fun platform(): String = "neoforge"
    override fun isModLoaded(modId: String): Boolean = ModList.get().isLoaded(modId)
    override fun isDevelopmentEnvironment(): Boolean = !FMLEnvironment.production
    override fun getMinecraftVersionString(): String = FMLLoader.versionInfo().mcVersion()
    override fun getConfigDir(): Path = FMLPaths.CONFIGDIR.get()
}